tests/test_code_hasher.py

```diff
@@ -0,0 +1,126 @@
+import pytest
+import threading
+
+from bionic.code_hasher import CodeHasher, TypePrefix
+
+
+def test_code_hasher():
+    def barray(value):
+        return bytearray(value, "utf8")
+
+    circular_dict_1_a = {"k11": "v1"}
```

Comment:
super nit: It seems a little weird that 1a, 2a, 3a are all part of one dict, while 1b, 2b, and 3b are all part of another. I'd have expected it to be 1a, 1b, 1c and 2a, 2b, 2c.